src/views/Entities/InternalServices.vue

```diff
@@ -0,0 +1,383 @@
+        size: this.pageSize,
+        offset: this.pageOffset
+      }
+
+      const endpoint = () => {
+        if (mesh === 'all') {
+          return this.$api.getAllServiceInsights(params)
+        } else if ((query && query.length) && mesh !== 'all') {
+          return this.$api.getServiceInsight(mesh, query, params)
+        }
+
+        return this.$api.getAllServiceInsightsFromMesh(mesh)
+      }
+
+      const getInternalServices = () => {
+        return endpoint()
+          .then(response => {
+            const items = () => {
+              const r = response
+
+              if ('total' in r) {
+                if (r.total !== 0 && r.items && r.items.length > 0) {
+                  return this.sortEntities(r.items)
+                }
+
+                return null
+              }
+
+              return r
+            }
+
+            const entityList = items()
+
+            if (items()) {
+              const firstItem = query
+                ? entityList
+                : entityList[0]
+
+              // set the first item as the default for initial load
+              this.firstEntity = firstItem.name
+
+              // load the YAML entity for the first item on page load
+              this.getEntity(stripTimes(firstItem))
+
+              // set the selected table row for the first item on page load
+              this.$store.dispatch('updateSelectedTableRow', firstItem.name)
+
+              this.tableData.data = query
+                ? [entityList]
+                : entityList
+
+              if (response.next) {
+                this.next = getOffset(response.next)
+                this.hasNext = true
+              } else {
+                this.hasNext = false
+              }
+
+              this.tableData.data = this.tableData.data.map(entity => {
+                // API can skip the field if there is no stat
+                entity.offline = entity.offline || 0
+                entity.online = entity.online || 0
+                entity.total = entity.total || 0
+                entity.totalOnline = `${entity.online} / ${entity.total}`
+
+                return entity
+              })
+
+              this.tableDataIsEmpty = false
+              this.isEmpty = false
+            } else {
+              this.tableData.data = []
+              this.tableDataIsEmpty = true
+              this.isEmpty = true
+
+              this.getEntity(null)
+            }
+          })
+          .catch(error => {
+            this.hasError = true
+            this.isEmpty = true
+
+            console.error(error)
+          })
+          .finally(() => {
+            setTimeout(() => {
+              this.isLoading = false
+            }, process.env.VUE_APP_DATA_TIMEOUT)
+          })
+      }
+
+      getInternalServices()
+    },
+    getEntity (entity) {
+      this.entityIsLoading = true
+      this.entityIsEmpty = false
+
+      const mesh = this.$route.params.mesh
+
+      if (entity && entity !== null) {
```

Comment:
second part is unnecessary as it's impossible to meet the first expression and not meet the next one